pkg/tls/self_signed_cert.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package gencert generates self-signed TLS certificates.
+package selfsignedcert
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"os"
+	"time"
+)
+
+const (
+	organization    = "F5 Inc."
+	filePermissions = 0o600
+)
+
+// nolint: revive
+func GenerateSelfSignedCert(certPath, keyPath string) (*tls.Certificate, error) {
+	// Return existing cert if cert exists
+	if _, certErr := os.Stat(certPath); certErr == nil {
+		if _, keyErr := os.Stat(keyPath); keyErr == nil {
+			cert, err := ReturnExistingCert(certPath, keyPath)
+			if err != nil {
+				return nil, err
+			}
+
+			return cert, nil
```

Comment:
```suggestion
			return ReturnExistingCert(certPath, keyPath)
```

Don't think there is a need for the additional err check here